webapp/javascript/ui/Input.tsx

```diff
@@ -1,41 +1,52 @@
-import React from 'react';
+import React, { Ref, ChangeEvent } from 'react';
 import { DebounceInput } from 'react-debounce-input';
 import styles from './Input.module.scss';
 
 interface InputProps {
   testId?: string;
   className?: string;
-  type: 'search';
+  type: 'search' | 'text' | 'password' | 'email' | 'number';
   name: string;
-  placeholder: string;
-  minLength: number;
-  debounceTimeout: number;
-  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  minLength?: number;
+  debounceTimeout?: number;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
   value: string | number;
+  htmlId?: string;
 }
 
-export default function Input({
-  testId,
-  className,
-  type,
-  name,
-  placeholder,
-  minLength,
-  debounceTimeout,
-  onChange,
-  value,
-}: InputProps) {
-  return (
-    <DebounceInput
-      data-testid={testId}
-      className={`${styles.input} ${className || ''}`}
-      type={type}
-      name={name}
-      placeholder={placeholder}
-      minLength={minLength}
-      debounceTimeout={debounceTimeout}
-      onChange={onChange}
-      value={value}
-    />
-  );
-}
+const Input = React.forwardRef(
```

Comment:
Why change this to `React.forwardRef`

Reply:
Here is the answer: https://stackoverflow.com/questions/58578570/value-of-using-react-forwardref-vs-custom-ref-prop
I used this approach because there are inputs with forwarded ref as `<input ref={ref}>` directly. But if you use input as functional standalone component `<Input ref={ref} />` doesn't work (`<Input customRef={ref} />` works). But, no problem, if you don't like this approach I can pass refs as <Input inputRef={ref} />, just let me know